hoth_api.yaml

```diff
@@ -1,62 +1,90 @@
 openapi: 3.0.2
 info:
   description: >-
-    This is the ATAT Provisioning API, an internal API solely meant for use by the ServiceNow component of ATAT in
-    order to kick off provisioning jobs.
-  version: '0.1'
-  title: ATAT Internal API - Provisioning
+    This is the HOTH (Hyperscaler Orchestration and Tracking Helper) API, a private internal-facing API that is intended
+    to be purely consumed by the ServiceNow component of ATAT
+  version: "0.1"
+  title: ATAT HOTH API - ServiceNow Integration
   contact:
     email: replaceme@ccpo.mil
 tags:
   - name: provisioning
     description: >-
       Operations related to provisioning jobs
+  - name: document
+    description: >-
+      Operations related to document generation requests
 paths:
+  /generate-document:
+    post:
+      tags:
+        - document
+      description: Generates a document
+      operationId: generateDocument
+      responses:
+        "200":
+          description: Successful operation
+          content:
+            application/pdf:
+              schema:
+                type: string
+                format: binary
+        "400":
+          description: Invalid input
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/ValidationError"
```

Comment:
We probably need a separate error type here too. Something like:

```json
{
  "errorCode": "ResponseTooLarge",
  "details": {
    "responseSize": 10202323,
    "maxAllowed": 6291456,
  }
}
```

We _know_ we have a 6MB response limit. And we don't currently know what our solution will be. Encoding a response for this would give us two benefits:
1. It's a code path we have to consider out of the gate in the first place
2. The client already has to anticipate a separate code path for this scenario (so that some day we can send a 201/303 response)

Though, this may be better as a `5xx` family error since there's nothing the client really did wrong (besides sending us too much text).

Should the DoW ever be >6MB? Probably not. But we need to anticipate hard limits of they're not part of our API.

Reply:
Should we return **413 Payload Too Large** as the error code?
https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/413

Reply:
Good call, and we should also remember to keep this in mind as we consider any potential solutions in the future. That is, whether one library or another, or one style or another (procedural vs template-driven) yields significantly smaller files.

Reply:
Looking into this more, Zach and I found that we probably want `422 Unprocessable Entity` if the request is otherwise correct